### ISSUE TITLE
2FA via mobile app

### ISSUE THREAD

**Issue by jwilk** [jwilk/mbank-cli#72]
Currently the only 2FA supported in mbank-cli is SMS OTP.
We should support mobile app authentication too.

Unfortunately, I don't have a phone capable of running the app, so I have no means to research this at the moment. :-(

(Cc: @chilek)

**Comment by chilek** [comment]
I've analyzed in google chrome web inspector - the difference is that after `initprepare` browser send request as below every second (in curl request format):
```bash
curl 'https://online.mbank.pl/api/auth/status' -H 'data-type: json' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36' -H 'Sec-Fetch-Mode: cors' -H 'content-type: application/json; charset=utf-8' -H 'x-request-verification-token: t+pXvUMl65EVnX274x50Kwwl9jvUqabXR1ggPfKvvr8uRWcqwZQdUHZo/NUG70kUMPePvpdPmzexYd/fJSfq0tAacKfkqHFXqPiDHF2xzFdr40IPtZP2V92AASv8pw9ICgJrQbxKWf8JmIF9os90DhzST8bYAzXqFCCMdpVozcpa/A2Vz+E3adRLyimLACPlcd2iYcTWN5IcpFT84iIzjoTEdomf1MciUvROHY8zo5+mmIbNUt/GF2PF0+t3' -H 'Referer: https://online.mbank.pl/authorization/onetime/set' -H 'x-requested-with: XMLHttpRequest' -H 'x-tab-id: edd9cb10-7218-4251-abc3-7ecde11fd93a' --data-binary '{"TranId":"4aecff94-ece9-457f-a775-23e5f50d094f"}' --compressed
```
and until there is not confirm click in mobile app get:
```
{"Status":"Prepared","Message":"","ErrorCode":""}
```
as reply.

**Comment by chilek** [comment]
And `execute` sent a little bit later looks as below:
```bash
curl 'https://online.mbank.pl/api/auth/execute' -H 'data-type: json' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36' -H 'Sec-Fetch-Mode: cors' -H 'content-type: application/json; charset=utf-8' -H 'x-request-verification-token: t+pXvUMl65EVnX274x50Kwwl9jvUqabXR1ggPfKvvr8uRWcqwZQdUHZo/NUG70kUMPePvpdPmzexYd/fJSfq0tAacKfkqHFXqPiDHF2xzFdr40IPtZP2V92AASv8pw9ICgJrQbxKWf8JmIF9os90DhzST8bYAzXqFCCMdpVozcpa/A2Vz+E3adRLyimLACPlcd2iYcTWN5IcpFT84iIzjoTEdomf1MciUvROHY8zo5+mmIbNUt/GF2PF0+t3' -H 'Referer: https://online.mbank.pl/authorization/onetime/set' -H 'x-requested-with: XMLHttpRequest' -H 'x-tab-id: edd9cb10-7218-4251-abc3-7ecde11fd93a' --data-binary '{}' --compressed
```

**Comment by chilek** [comment]
Please add to mbank-cli cyclic `/api/auth/status' check and displaying result on the screen so I will be able to debug it and reply you with response content dump after confirm click in mobile app. Google Chrome shows non-zero response Content-Length, but I'm not able to look at confirmed status request response data!

**Comment by chilek** [comment]
`TranId` is taken from `initprepare` json response.

**Comment by chilek** [comment]
Ok I've caught `status` response after clicking in mobile app:
First:
```json
{"Status":"PreAuthorized","Message":"","ErrorCode":""}
```
and then:
```json
{"Status":"Authorized","Message":"","ErrorCode":""}
```

**Comment by chilek** [comment]
In cancel/deny scenario played in mobile app we get the following replies to status requests:
1.
```json
{"Status":"PreAuthorized","Message":"","ErrorCode":""}
```
2. 
```json
{"Status":"Canceled","Message":"","ErrorCode":""}
```

**Comment by chilek** [comment]
Please merge my changes ;-)

**Comment by jwilk** [comment]
Fixed in 667938ad185e99eea8c9b7f8af91e693dfefd6e0.

I'll make a new release later today.

**Comment by chilek** [comment]
Thanks!

**Comment by jwilk** [comment]
Fixed in [mbank-cli 2.1](https://github.com/jwilk/mbank-cli/releases/tag/2.1).
